File: backend/models/Issue.js
const mongoose = require("mongoose");

const IssueSchema = new mongoose.Schema(
  {
    title: { 
      type: String, 
      required: true 
    },

    description: { 
      type: String 
    },

    severity: { 
      type: Number, 
      required: true 
    },

    frequency: { 
      type: Number, 
      required: true 
    },

    userImpact: { 
      type: Number, 
      required: true 
    },

    priorityScore: { 
      type: Number, 
      default: 0 
    },

    status: { 
      type: String, 
      required: true 
    }
  },
  { timestamps: true }
);

module.exports = mongoose.model("Issue", IssueSchema);

File: priority-engine-java/Issue.java
public class Issue {
    public int severity;
    public int frequency;
    public int userImpact;

    public Issue(int severity, int frequency, int userImpact) {
        this.severity = severity;
        this.frequency = frequency;
        this.userImpact = userImpact;
    }

    public int calculatePriority() {
        return severity * 3 + frequency * 2 + userImpact;
    }
}
